code/game/barricade.dm

```diff
@@ -0,0 +1,980 @@
+	var/stack_amount = 5 //The amount of stack dropped when disassembled at full health
+	var/destroyed_stack_amount //to specify a non-zero amount of stack to drop when destroyed
+	max_integrity = 100 //Pretty tough. Changes sprites at 300 and 150
+	var/crusher_resistant = TRUE //Whether a crusher can ram through it.
+	var/base_acid_damage = 2
+	var/barricade_resistance = 5 //How much force an item needs to even damage it at all.
+	var/barricade_hitsound
+
+	var/barricade_type = "barricade" //"metal", "plasteel", etc.
+	var/can_change_dmg_state = TRUE
+	var/damage_state = 0
+	var/closed = FALSE
+	var/can_wire = FALSE
+	var/is_wired = FALSE
+	var/image/wired_overlay
+	flags_barrier = HANDLE_BARRIER_CHANCE
+
+/obj/structure/barricade/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/structure/barricade/handle_barrier_chance(mob/living/M)
+	return prob(max(30,(100.0*obj_integrity)/max_integrity))
+
+/obj/structure/barricade/examine(mob/user)
+	..()
+
+	if(is_wired)
+		to_chat(user, "<span class='info'>There is a length of wire strewn across the top of this barricade.</span>")
+	switch(damage_state)
+		if(0) to_chat(user, "<span class='info'>It appears to be in good shape.</span>")
+		if(1) to_chat(user, "<span class='warning'>It's slightly damaged, but still very functional.</span>")
+		if(2) to_chat(user, "<span class='warning'>It's quite beat up, but it's holding together.</span>")
+		if(3) to_chat(user, "<span class='warning'>It's crumbling apart, just a few more blows will tear it apart.</span>")
+
+/obj/structure/barricade/hitby(atom/movable/AM)
+	if(AM.throwing && is_wired)
+		if(iscarbon(AM))
+			var/mob/living/carbon/C = AM
+			C.visible_message("<span class='danger'>The barbed wire slices into [C]!</span>",
+			"<span class='danger'>The barbed wire slices into you!</span>")
+			C.apply_damage(10)
+			C.KnockDown(2) //Leaping into barbed wire is VERY bad
+	..()
+
+
+
+/obj/structure/barricade/Bumped(atom/A)
+	..()
+
+	if(istype(A, /mob/living/carbon/xenomorph/crusher))
+
+		var/mob/living/carbon/xenomorph/crusher/C = A
+
+		if(C.charge_speed < CHARGE_SPEED_MAX * 0.5)
+			return
+
+		if(crusher_resistant)
+			obj_integrity -= C.charge_speed * CRUSHER_CHARGE_BARRICADE_MULTI
+			update_health()
+
+		else if(!C.stat)
+			visible_message("<span class='danger'>[C] smashes through [src]!</span>")
+			destroy_structure()
+
+/obj/structure/barricade/CheckExit(atom/movable/O, turf/target)
+	if(closed)
+		return TRUE
+
+	if(O.throwing)
+		if(is_wired && iscarbon(O)) //Leaping mob against barbed wire fails
+			if(get_dir(loc, target) & dir)
+				return FALSE
+		return TRUE
+
+	if(get_dir(loc, target) & dir)
+		return FALSE
+	else
+		return TRUE
+
+/obj/structure/barricade/CanPass(atom/movable/mover, turf/target)
+	if(closed)
+		return TRUE
+
+	if(mover && mover.throwing)
+		if(is_wired && iscarbon(mover)) //Leaping mob against barbed wire fails
+			if(get_dir(loc, target) & dir)
+				return FALSE
+		return TRUE
+
+	if(istype(mover, /obj/vehicle))
```

Comment:
This needs removing, it's referring to a vehicle var that we don't have

Reply:
I think I have that now in change commit 746e6d4